### PR TITLE
Build with production-ready React

### PR DESCRIPTION
**Pull request by katiefenn**
- Introduced a WebPack Define plugin to remove development code from bundle

### CODE REVIEW

**Comment by katiefenn** [comment]
Bundle size stats:

With development code: 476kb
With development code (gzipped): 138kb
Without development code: 388kb
Without development code (gzipped): 114kb

**Comment by katiefenn** [comment]
Warnings in the console have gone:

screen shot 2016-10-05 at 10 39 33

File: webpack/prod-config.js [hunk 1]
```diff
@@ -31,7 +31,12 @@ module.exports = {
...
       }
     }),
-    webpackIsomorphicToolsPlugin
+    webpackIsomorphicToolsPlugin,
+    new webpack.DefinePlugin({
+        'process.env': {
+          'NODE_ENV': JSON.stringify('production')
+        }
+      })
   ],
```
Comment (katiefenn) [inline webpack/prod-config.js]: The WebPack plugin docs explain how this works nicely:

The values will be inlined into the code which allows a minification pass to remove the redundant conditional.

Example:
```
if (!PRODUCTION)
    console.log('Debug info')
if (PRODUCTION)
    console.log('Production log')
```
After passing through webpack with no minification results in:
```
if (!true)
    console.log('Debug info')
if (true)
    console.log('Production log')
```
and then after a minification pass results in:
```
console.log('Production log')
```

File: webpack/prod-config.js [hunk 2]
```diff
@@ -31,7 +31,12 @@ module.exports = {
         warnings: false
       }
     }),
```
Comment (katiefenn) [inline webpack/prod-config.js]: Further reading on preparing react for production: http://moduscreate.com/optimizing-react-es6-webpack-production-build/

**Comment by earnubs** [comment]
[Minimise bundle size](https://trello.com/c/TrpkGlR3/82-minimise-bundle-size)

**Pull request merged**